docs/content/concepts/partitions-schedules-sensors/partitioning-assets.mdx

```diff
@@ -166,8 +167,8 @@ def image_sensor(context: SensorEvaluationContext):
     new_images = [
         img_filename
         for img_filename in os.listdir(os.getenv("MY_DIRECTORY"))
-        if not context.instance.has_dynamic_partition(
-            images_partitions_def.name, img_filename
+        if not images_partitions_def.has_partition_key(
```

Comment:
@clairelin135  interested in your opinion if this is an ok pattern to suggest. The issue in the old snippet was that `images_partitions_def.name` is an `Optional[str]`, but `context.instance.has_dynamic_partition` is expecting a `str`. Switching to this method bypasses the optionality issue.

Reply:
yup, this looks reasonable to me